lib/Parse/ParseDecl.cpp

```diff
@@ -970,7 +972,20 @@ bool Parser::parseDifferentiableAttributeArguments(
   SyntaxParsingContext ContentContext(
       SyntaxContext, SyntaxKind::DifferentiableAttributeArguments);
 
-  // Parse optional differentiation parameters, starting with the 'wrt:' label.
+  // Parse optional differentiation parameters.
+  // Parse 'linear' label (optional).
+  if (Tok.is(tok::identifier) && Tok.getText() == "linear") {
+    linear = true;
+    consumeIdentifier();
```

Comment:
```suggestion
    consumeIdentifier(tok::identifier);
```

This makes an assertion about the identifier being consumed, preventing errors in future changes just in case.

Reply:
`consumeIdentifier()` does the following assertion inside to check if it's `identifier`, but also does a couple more:
```c++
assert(Tok.isAny(tok::identifier, tok::kw_self, tok::kw_Self));
```
However, I think calling `consumeToken(tok::identifier)` could be a better option as it internally does the following assertion to specifically only check if it's an `identifier`:
```c++
SourceLoc consumeToken(tok K) {
  assert(Tok.is(K) && "Consuming wrong token kind");
  return consumeToken();
}
```

Reply:
Yes, it should be `consumeToken()`. I didn't notice that and thought you wrote `consumeToken` already :)